kingpin/actors/rightscale/server_array.py

```diff
@@ -328,42 +416,32 @@ def _execute(self):
         yield self._client.login()
 
         # First, find the array we're going to be terminating.
-        self.array = yield self._find_server_arrays(self.option('array'),
-                                                    raise_on='notfound',
-                                                    allow_mock=False)
+        arrays = yield self._find_server_arrays(self.option('array'),
+                                                raise_on='notfound',
+                                                allow_mock=False,
+                                                exact=self.option('exact'))
 
         # Disable the array so that no new instances launch. Ignore the result
         # of this opertaion -- as long as it succeeds, we're happy. No need to
         # store the returned server array object.
-        params = self._generate_rightscale_params(
-            'server_array', {'state': 'disabled'})
-        if not self._dry:
-            self.log.info('Disabling Array "%s"' % self.option('array'))
-            yield self._client.update_server_array(self.array, params)
-        else:
-            self.log.info('Would have updated array "%s" with params: %s' %
-                          (self.option('array'), params))
+        yield self._act_on_arrays(self._disable_array, arrays)
```

Comment:
since this is an array-specific actor, I'd recommend naming this `self._apply(<function>, <arrays>)`